tests/instruments/hp/test_hp856Xx.py

```diff
@@ -0,0 +1,477 @@
+
+
+class TestHP856Xx:
+    def test_id(self):
+        with expected_protocol(
+                HP856Xx,
+                [("ID?", "HP8560A,002,H03")],
+        ) as instr:
+            assert instr.id == "HP8560A,002,H03"
+
+    def test_attenuation(self):
+        with expected_protocol(
+                HP856Xx,
+                [("AT 70", None),
+                 ("AT?", "70"),
+                 ("AT AUTO", None),
+                 ("AT?", "20"),
+                 ("AT 20", None),
+                 ("AT?", "20")],
+        ) as instr:
+            # test set and get of attenuation as integer
+            instr.attenuation = 70
+            assert instr.attenuation == 70
+            # test string parameters
+            instr.attenuation = "AUTO"
+            assert instr.attenuation == 20
+            # test truncation
+            instr.attenuation = 16
+            assert instr.attenuation == 20
+
+    @pytest.mark.parametrize("amplitude_unit", [e for e in AmplitudeUnits])
+    def test_amplitude_units(self, amplitude_unit):
+        with expected_protocol(
+                HP856Xx,
+                [("AUNITS " + amplitude_unit, None),
+                 ("AUNITS?", amplitude_unit)],
+        ) as instr:
+            instr.amplitude_unit = amplitude_unit
+            assert instr.amplitude_unit == amplitude_unit
+
+    @pytest.mark.parametrize(
+        "function,command",
+        [
+            ("auto_couple", "AUTOCPL"),
+            ("exchange_traces", "AXB"),
+            ("subtract_display_line_from_trace_b", "BML"),
+            ("continuous_sweep", "CONTS"),
+            ("full_span", "FS"),
+            ("linear_scale", "LN"),
+            ("marker_to_center_frequency", "MKCF"),
+            ("marker_minimum", "MKMIN")
+        ]
+    )
+    def test_primitive_commands(self, command, function):
+        """
+        Tests primitive commands which have no parameter or query derivat
+        """
+        with expected_protocol(
+                HP856Xx,
+                [(command, None)]
+        ) as instr:
+            getattr(instr, function)()
+
+    @pytest.mark.parametrize("trace", [e for e in Trace])
+    def test_blank_trace(self, trace):
+        with expected_protocol(
+                HP856Xx,
+                [("BLANK " + trace, None)]
+        ) as instr:
+            instr.blank_trace(trace)
+
+    def test_blank_trace_exceptions(self):
+        with expected_protocol(
+                HP856Xx,
+                []
+        ) as instr:
+            with pytest.raises(ValueError):
+                instr.blank_trace("TEST")
+
+            with pytest.raises(TypeError):
+                instr.blank_trace(0)
+
+    @pytest.mark.parametrize("function,command", [
+        ("start_frequency", "FA"),
+        ("center_frequency", "CF"),
+        ("stop_frequency", "FB"),
+        ("frequency_offset", "FOFFSET")
+    ])
+    @pytest.mark.parametrize("hp_derivat,max_freq", [(HP8560A, 2.9e9), (HP8561B, 6.5e9)])
```

Comment:
According to normal python rules, I'd add a space:
```suggestion
    @pytest.mark.parametrize("hp_derivat, max_freq", [(HP8560A, 2.9e9), (HP8561B, 6.5e9)])
```